File: c++/17619.cpp
#include <iostream>

using namespace std;

int arr[13];
int sum = 0;
int cnt = 0;
int k;
bool visited[3000000];

void recur(int n, int curr) {
  if (curr >= 1 && !visited[curr]) {
    cnt++;
    visited[curr] = true;
  }

  if (n == k) return;

  recur(n + 1, curr);
  recur(n + 1, curr - arr[n]);
  recur(n + 1, curr + arr[n]);
}

int main() {
  freopen("example.txt", "r", stdin);
  cin >> k;

  for (int i = 0; i < k; i++) {
    cin >> arr[i];
    sum += arr[i];
  }

  recur(0, 0);

  cout << sum - cnt;

  return 0;
}

File: js/17619.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const k = +inputs[0];
const arr = inputs[1].split(' ').map(Number);
const sum = arr.reduce((a, b) => a + b, 0);
const isVisited = Array(3000000).fill(false);
let cnt = 0;

recur(0, 0);

console.log(sum - cnt);

function recur(n, curr) {
  if (curr >= 1 && !isVisited[curr]) {
    cnt++;
    isVisited[curr] = true;
  }

  if (n === k) return;

  recur(n + 1, curr);
  recur(n + 1, curr + arr[n]);
  recur(n + 1, curr - arr[n]);
}
